### PR TITLE
Migrating from dep to gomod

### PR DESCRIPTION
**Pull request by jayaramsatya**
1. Added newer version cni spec 0.4.0
2. Make changes are done to remove building vendor
3. Upgraded the kubernetes client package to version 1.17

### CODE REVIEW

File: cmd/opflexagentcni/main.go [hunk 1]
```diff
@@ -285,9 +285,45 @@ func cmdDel(args *skel.CmdArgs) error {
...
+			return err
+		}
+	}
+	if n.NetConf.RawPrevResult == nil {
+		return fmt.Errorf("Required prevResult missing")
+	}
+	if err := version.ParsePrevResult(&n.NetConf); err != nil {
+		return err
+	}
+	result, err := current.NewResultFromResult(n.PrevResult)
```
Comment (kishorejonnala) [inline cmd/opflexagentcni/main.go:312]: Thanks for taking care of this PrevResult part.

File: .travis.yml [hunk 2]
```diff
@@ -3,9 +3,8 @@ matrix:
   include:
     - language: go
       sudo: false
-      go: "1.10"
+      go: "1.13"
```
Comment (snaiksat) [inline .travis.yml:6]: We need to confirm that the base container image with which we ship (i.e. UBI) supports 1.13.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	k8s.io/api v0.17.3
+	k8s.io/apimachinery v0.17.3
+	k8s.io/client-go v8.0.0+incompatible
+	k8s.io/kubernetes v0.0.0-00010101000000-000000000000
+	sigs.k8s.io/controller-runtime v0.5.0
+)
+
+replace (
+	go.etcd.io/bbolt => go.etcd.io/bbolt v1.3.3
+	k8s.io/api => k8s.io/kubernetes/staging/src/k8s.io/api v0.0.0-20200211180626-06ad960bfd03
```
Comment (snaiksat) [inline go.mod:85]: After doing this, and some of the other version pining below, to what extent will we be compatible with an older version of kubernetes API/server? There will be users who will have deployed k8s < 1.17, either directly, or by virtue of using a distro like OpenShift/RH, and we wont be able to force them to upgrade their k8s deployment.

Comment (jayaramsatya) [reply go.mod:85]: Sumit, it should be backward compatible, may be we should validate, if we have any 1.13 older version fabs

File: .travis.yml (hunk 2 above)
Comment (jayaramsatya) [reply .travis.yml:6]: Go modules full support introduced from go 1.12 on-words.

File: go.mod (hunk 3 above)
Comment (jayaramsatya) [reply go.mod:85]: BTW Currently we are using 1.15 API version

File: .travis.yml (hunk 2 above)
Comment (snaiksat) [reply .travis.yml:6]: I am not referring to gomod, I am referring to the golang version 1.13, and verifying it being available in the UBI base container version that we intend to use. For instance, RHEL 7 did not have golang 1 .13.

File: go.mod (hunk 3 above)
Comment (snaiksat) [reply go.mod:85]: Why just 1.13? Elsewhere below you have version pinned to 1.17. So what about any version < 1.17? We will have fewer, perhaps if any customers on, say 1.13, but we will definitely customers on 1.16 (also need to confirm what OpenShift 3.11 has).

**Pull request merged**